File: frontEnd/src/types/dto/stepDTO/KeyboardStepDTO.ts
// 键盘操作
import {StepBaseDTO} from "@/types/dto/base/StepBaseDTO";
import {KeyboardInputType} from "@/types/enum/keyboard/KeyboardInputType";
import {KeyboardKey} from "@/types/enum/keyboard/KeyboardKey";
import {ElementDTO} from "@/types/dto/ElementDTO";

export class KeyboardStepDTO extends StepBaseDTO {
    stepType: string = 'KEYBOARD'; // 步骤类型
    element: ElementDTO = new ElementDTO(); // 操作元素
    inputType: KeyboardInputType | string = 'NORMAL'; // 输入类型

    keyboardKey?: KeyboardKey | string = "ENTER"; // 按键
    inputValue?: string = ''; // 输入内容
    isAdditional?: number = 0; // 是否追加输入
}

File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/domain/ui/dto/step/KeyboardStepDTO.java
package com.mokatest.platform.demos.domain.ui.dto.step;// 键盘操作


import com.mokatest.platform.demos.domain.ui.uiEnum.keyboard.KeyboardInputType;
import com.mokatest.platform.demos.domain.ui.uiEnum.keyboard.KeyboardKey;
import com.mokatest.platform.demos.domain.ui.dto.ElementDTO;
import lombok.Data;

@Data
public class KeyboardStepDTO extends StepBaseDTO {
    private String stepType; // 步骤类型
    private ElementDTO element; // 操作元素
    private KeyboardInputType inputType; // 输入类型

    private KeyboardKey keyboardKey; // 按键
    private String inputValue; // 输入内容
    private Integer isAdditional; // 是否追加输入
}